brave/app/scripts/metamask-controller.js

```diff
@@ -0,0 +1,14 @@
+const MetamaskController = require('../../../app/scripts/metamask-controller')
+const nodeify = require('../../../app/scripts/lib/nodeify')
+
+module.exports = class BraveController extends MetamaskController {
+  constructor (opts) {
+    super(opts)
+  }
+
+  getApi () {
+    const api = super.getApi()
+    api['setBatTokenAdded'] = nodeify(this.preferencesController.setBatTokenAdded, this.preferencesController)
```

Comment:
nit: I think  `api.setBatTokenAdded` is preferred here.

Reply:
done